content-src/asrouter/templates/FirstRun/FirstRun.jsx

```diff
@@ -0,0 +1,216 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from "react";
+import { Interrupt } from "./Interrupt";
+import { Triplets } from "./Triplets";
+import { actionCreators as ac, actionTypes as at } from "common/Actions.jsm";
+import { addUtmParams } from "./addUtmParams";
+
+export const FLUENT_FILES = [
+  "branding/brand.ftl",
+  "browser/branding/brandings.ftl",
+  "browser/branding/sync-brand.ftl",
+  "browser/newtab/onboarding.ftl",
+];
+
+export const helpers = {
+  selectInterruptAndTriplets(message = {}) {
+    const hasInterrupt = Boolean(message.content);
+    const hasTriplets = Boolean(message.bundle && message.bundle.length);
+    const UTMTerm = message.utm_term || "";
+    return {
+      hasTriplets,
+      hasInterrupt,
+      interrupt: hasInterrupt ? message : null,
+      triplets: hasTriplets ? message.bundle : null,
+      UTMTerm,
+    };
+  },
+
+  addFluent(document) {
+    FLUENT_FILES.forEach(file => {
+      const link = document.head.appendChild(document.createElement("link"));
+      link.href = file;
+      link.rel = "localization";
+    });
+  },
+
+  async fetchFlowParams({ fxaEndpoint, UTMTerm, dispatch, setFlowParams }) {
+    try {
+      const url = new URL(
+        `${fxaEndpoint}/metrics-flow?entrypoint=activity-stream-firstrun&form_type=email`
+      );
+      addUtmParams(url, UTMTerm);
+      const response = await fetch(url, { credentials: "omit" });
+      if (response.status === 200) {
+        const { deviceId, flowId, flowBeginTime } = await response.json();
+        setFlowParams({ deviceId, flowId, flowBeginTime });
+      } else {
+        dispatch(
+          ac.OnlyToMain({
+            type: at.TELEMETRY_UNDESIRED_EVENT,
+            data: {
+              event: "FXA_METRICS_FETCH_ERROR",
+              value: response.status,
+            },
+          })
+        );
+      }
+    } catch (error) {
+      dispatch(
+        ac.OnlyToMain({
+          type: at.TELEMETRY_UNDESIRED_EVENT,
+          data: { event: "FXA_METRICS_ERROR" },
+        })
+      );
+    }
+  },
+};
+
+export class FirstRun extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.didLoadFlowParams = false;
+
+    this.state = {
```

Comment:
class properties = in order to not trigger a re-render

I mostly kept everything in state here so I could reuse `selectInterruptAndTriplets` from the hooks implementation. Additional renders aren't triggered because they're derived. I could refactor it further though